### ISSUE TITLE
Background style not-transitioning

### ISSUE THREAD

**Issue by egonelbre** [gopherjs/vecty#190]
For some reason in this example https://gist.github.com/egonelbre/acd3a064caf656cccd7a157a8bdcf6b3 the background is not transitioning properly.

When you remove `ItemView.Key` it does, however then the moving item doesn't change the color, but rather the content is moved around.

**Comment by pdf** [comment]
This appears to be behaving as expected - by implementing the `vecty.Keyer` interface, you allow vecty to _move_ elements around in a list, rather than mutate existing elements in place.  When dealing with complex elements this can be a big saving, as it avoids tearing down incompatible elements and rebuilding them when an earlier sibling is removed.  Because the browser moves the element intact, there's no transition, the element simply moves.

Does that make sense?

**Comment by egonelbre** [comment]
Yes, mounting and reuse behavior is correct however the background color changes depending on the index, which has a 1s transition.

Removing the Key makes the transition work however it breaks the correct mapping, as you mentioned.

**Comment by egonelbre** [comment]
Maybe this makes it more obvious https://gist.github.com/egonelbre/c71fdb90955c599e6f1b8007bf6c6af9 ... the width of the "progress-bar" should change smoothly with an animation.

**Comment by pdf** [comment]
If you use a `vecty.List` instead of a slice, I think it does what you expect?

https://gist.github.com/pdf/c689fa08b585197e917a3d382acdbe21

This is currently the recommended data type for containing dynamic lists of elements.  I suspect there may still be a bug or improvement we can make here somewhere, but it's been some time since I've looked hard at this code, and we've not had much time to work on this project in recent months.
